posts/frangeris/03.secure-values-by-private-keys-in-aws.md

```diff
@@ -0,0 +1,235 @@
+---
+title: Ensuring secure values by private keys in AWS (KMS, SSM, Secrets Manager)
+published: false
+description: Secure way of store and access sensible values on AWS
+tags: "aws, private keys security, serverless security, cloud security"
+cover_image: ./assets/03.cover.png
+---
+
+## Ensuring secure values by private keys in AWS (KMS, SSM, Secrets Manager)
+
+Private keys, credentials access and sensitive information such as passwords or any kind of confidential data are essentials in applications logic when interacting with other services or even databases, storing those values can be risky if proper security measures are not in place.
+
+In this post we will dig into one way to accomplish secure access to sensible information in AWS environments. For simplicity we will be using:
+
+- A ready to use [AWS account](https://aws.amazon.com/).
+- [Serverless Framework](https://www.serverless.com/framework/docs/getting-started) - to delegate the CloudFormation creation and deployment.
+
+> This implementation focuses on using specifics services and functionalities provided by AWS services. Implementations will vary depending on the provider (Azure, GCP).
+
+### Let's start by creating a basic [AWS Lambda](https://aws.amazon.com/lambda/)
+
+This function will try to access our "secure" and use it. By using `npx` command we get access to serverless executable, select the basic to keep it simple, there are options for different types of uses cases, even different languages options in case `Node.js` is not your language:
+
+![Run npx serverless command to start](./assets/03.image.1.png)
+
+Some operations used on the lambda function will require some permissions later, due to this, it's recommended to include and enable those permissions before the IAM role of the lambda gets created, here is what permissions we need the lambda to have:
+
+```yaml
+provider:
+  # ...
+  iam:
+    role:
+      statements:
+        - Effect: Allow
+          Action:
+            - "s3:*"
+          Resource: "*"
+        - Effect: Allow
+          Action:
+            - "cloudwatch:*"
+            - "logs:*"
+          Resource: "*"
+        - Effect: Allow
+          Action:
+            - "ssm:GetParameter"
+            - "ssm:GetParametersByPath"
+          Resource: "*"
+        - Effect: Allow
+          Action:
+            - "secretsmanager:GetSecretValue"
+          Resource: "*"
+        - Effect: Allow
+          Action:
+            - "kms:Decrypt"
+          Resource: "*"
+```
+
+Then, with your [aws account](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html) already set up, let's deploy it.
+
+```bash
+npx sls deploy
+```
+
+🎉 After a correct deployment, the code should be created successfully and ready to use, now we need to configure the most important part, security.
```

Comment:
```suggestion
🎉 After a correct deployment, the code should have been created successfully and be ready to use. Now we need to configure the most important part: security.
```

Reply:
The repetition of `be` here regards the different uses you give to it: the first one is for a verb and the second one for an adj. I believe it's clearer (and more correct) with a different tense